brainscore_core/plugin_management/parse_plugin_changes.py

```diff
@@ -0,0 +1,138 @@
+	depending on whether there are any changed non-plugin files.
+	"""
+	plugin_info_dict["is_plugin_only"] = "False" if num_changed_non_plugin_files > 0 else "True"
+
+
+def _plugin_name_from_path(path_relative_to_library: str) -> str:
+    """
+    Returns the name of the plugin from the given path. 
+    E.g. `_plugin_name_from_path("brainscore_vision/models/mymodel")` will return `"mymodel"`.
+    """
+    return path_relative_to_library.split('/')[2]
+
+
+def get_changed_plugin_paths(plugin_info_dict: dict, changed_plugin_files: List[str], domain_root: str):
+	"""
+	Adds full path (rel. to library) of all changed plugin directories for each plugin_type to plugin_info_dict
+	"""
+	plugin_info_dict["changed_plugins"] = {}
+	for plugin_type in PLUGIN_DIRS:
+		plugin_type_path = f'{domain_root}/{plugin_type}/'
+		changed_plugin_paths = [fpath for fpath in changed_plugin_files if fpath.startswith(plugin_type_path)]
+		plugin_info_dict["changed_plugins"][plugin_type] = list(set([_plugin_name_from_path(fname) 
+			for fname in changed_plugin_paths if f'/{plugin_type}/' in fname]))
+
+
+def _get_plugin_ids(plugin_type: str, new_plugin_dirs: List[str], domain_root: str) -> List[str]:
+	"""
+	Searches all __init.py__ files in `new_plugin_dirs` of `plugin_type` for registered plugins.
+	Returns list of identifiers for each registered plugin.
+	"""
+	plugin_ids = []
+
+	for plugin_dirname in new_plugin_dirs:
+		init_file = Path(f'{domain_root}/{plugin_type}/{plugin_dirname}/__init__.py')
+		with open(init_file) as f:
+			registry_name = plugin_type.strip(
+				's') + '_registry'  # remove plural and determine variable name, e.g. "models" -> "model_registry"
+			plugin_registrations = [line for line in f if f"{registry_name}["
+									in line.replace('\"', '\'')]
+			for line in plugin_registrations:
+				result = re.search(f'{registry_name}\[.*\]', line)
+				identifier = result.group(0)[len(registry_name) + 2:-2] # remove brackets and quotes
+				plugin_ids.append(identifier)
+
+	return plugin_ids
+
+
+def parse_plugin_changes(commit_SHA: str, domain_root: str) -> dict:
+	"""
+	Return information about which files changed by the invoking PR (compared against main) belong to plugins
+
+	:param commit_SHA: SHA of the invoking PR
+	:param domain_root: the root package directory of the repo where the PR originates, either 'brainscore' (vision) or 'brainscore_language' (language)
+	"""
+	plugin_info_dict = {}
+	changed_files = get_all_changed_files(commit_SHA)
+	changed_plugin_files, changed_non_plugin_files = get_changed_plugin_files(changed_files)
+	is_automergeable(plugin_info_dict, len(changed_non_plugin_files))
+	get_changed_plugin_paths(plugin_info_dict, changed_plugin_files, domain_root)
+
+	return plugin_info_dict
+	
+
+def get_plugin_info(commit_SHA: str, domain_root: str):
+	"""
+	If any model or benchmark files changed, get plugin ids and set run_score to "True"
+	Otherwise set else "False"
```

Comment:
```suggestion
	If any model or benchmark files changed, get plugin ids and set run_score to "True".
	Otherwise set to "False".
	Print all collected information about plugins.
```